### ISSUE TITLE
Failure to generate an attribute with nested wildcarded generic type

### ISSUE THREAD

**Issue by ylexus** [immutables/immutables#797]
The following fails to compile with the error `Cannot parse type from input string 'java.util.function.Supplier<java.util.stream.Stream<<any>>>'. unexpected term '<'.`

Tried versions 2.4.4 and 2.6.1.

```
@Target({ElementType.PACKAGE, ElementType.TYPE})
@Retention(RetentionPolicy.CLASS) // Make it class retention for incremental compilation
@Value.Style(
        get = {"is*", "get*"}, // Detect 'get' and 'is' prefixes in accessor methods
        init = "set*", // Builder initialization methods will have 'set' prefix
        typeAbstract = "Base*", // 'Base' prefix will be detected and trimmed
        typeImmutable = "*", // No prefix or suffix for generated immutable type
        depluralize = true,  // Do not use plural in setter methods for adding single entry to maps and collections
        visibility = Value.Style.ImplementationVisibility.PUBLIC) // Generated class will be always public
public @interface ImmutablesStyle {}
```
```
@Value.Immutable
@ImmutablesStyle
final class BaseTempType<K, V> {
}
```
```
import java.util.function.Supplier;
import java.util.stream.Stream;

@Value.Immutable
@ImmutablesStyle
interface BaseSuppliedSourceTypeConfig extends SourceTypeConfig {
    @Value.Default
    default Supplier<Stream<TempType<?, ?>>> getEventSupplier() {
        return Stream::empty;
    }
}
```

Compiler stack trace (taken from IntelliJ IDEA error window):
```
Error:java: org.immutables.value.internal.$processor$.$Processor threw java.lang.IllegalArgumentException: Cannot parse type from input string 'java.util.function.Supplier<java.util.stream.Stream<<any>>>'. unexpected term '<'
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser$1Reader.type($Type.java:751)
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser$1Reader.arguments($Type.java:792)
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser$1Reader.type($Type.java:728)
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser$1Reader.arguments($Type.java:792)
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser$1Reader.type($Type.java:728)
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser.doParse($Type.java:852)
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser.parse($Type.java:700)
  	at org.immutables.value.internal.$processor$.meta.$FromSupertypesModel.isEligibleFromType($FromSupertypesModel.java:136)
  	at org.immutables.value.internal.$processor$.meta.$FromSupertypesModel.<init>($FromSupertypesModel.java:82)
  	at org.immutables.value.internal.$processor$.meta.$ValueType.getBuildFromTypes($ValueType.java:1228)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables._t18__generateBuilder($Generator_Immutables.java:4812)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables$FragmentDispatch.run($Generator_Immutables.java:16115)
  	at org.immutables.value.internal.$generator$.$Templates$Fragment.invoke($Templates.java:248)
  	at org.immutables.value.internal.$generator$.$Intrinsics.$($Intrinsics.java:96)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables._t4__generateImmutable($Generator_Immutables.java:579)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables$FragmentDispatch.run($Generator_Immutables.java:16101)
  	at org.immutables.value.internal.$generator$.$Templates$Fragment.invoke($Templates.java:248)
  	at org.immutables.value.internal.$generator$.$Intrinsics.$($Intrinsics.java:96)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables$1.run($Generator_Immutables.java:23)
  	at org.immutables.value.internal.$generator$.$Templates$Fragment.invoke($Templates.java:248)
  	at org.immutables.value.internal.$generator$.$Output$7.invoke($Output.java:163)
  	at org.immutables.value.internal.$generator$.$Intrinsics.$($Intrinsics.java:96)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables._t0__generate($Generator_Immutables.java:20)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables$FragmentDispatch.run($Generator_Immutables.java:16097)
  	at org.immutables.value.internal.$generator$.$Templates$Fragment.invoke($Templates.java:248)
  	at org.immutables.value.internal.$generator$.$AbstractGenerator.invoke($AbstractGenerator.java:57)
  	at org.immutables.value.internal.$processor$.$Processor.process($Processor.java:63)
  	at org.immutables.value.internal.$generator$.$AbstractGenerator.process($AbstractGenerator.java:87)
  	at org.immutables.processor.ProxyProcessor.process(ProxyProcessor.java:72)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:705)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
  	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
  	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
  	at com.sun.tools.javac.main.Main.compile(Main.java:523)
  	at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
  	at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
...
```

**Comment by elucash** [comment]
we've made certain fixes to parsing and also to discover proper types. This is special behavior (or bug) of javac, `TempType` is not yet generated, and combining with generics, the compiler gives up and produces type with `"<any>"` representation, which is then we cannot handle. Please, try with 2.6.3 or newly released 2.7.1 to see if fixes work. Thank you!

**Comment by ylexus** [comment]
Still the same with 2.7.3.

```
Warning:(11, 1) java: (immutables) Type parsing problem in FromSupertypesModel: java.lang.IllegalArgumentException: Cannot parse type from input string 'java.util.function.Supplier<java.util.stream.Stream<<any>>>'. unexpected term '<'
Warning:(11, 1) java: (immutables:from) Generated builder '.from' method will not copy from attribute 'eventSupplier' because it has different return type in supertype (And we cannot handle generic specialization or covariant overrides yet). Sometimes it is possible to avoid this by providing abstract override method in this value object
```

**Comment by elucash** [comment]
@ylexus thank you for feedback! As silly as it may sound, there I actually tried to solve the problem with the thrown exception, making (FromSupertypesModel) resistant to failures in parsing types.
But after taking another look, I actually was able to find why other machinery of "guessing" the type did not work.
The `TempType` here is not yet generated type and javac has such a bug (or deficiency in my view) that if such a type has any type parameter it suddenly becomes determined/printed as "<any>" instead. The type it is `TempType` is unresolved so javac cannot know its name, but rather than leave it unchanged in its string representation as `TempType<?, ?>` it becomes `<any>`, and this is not happening for types without generic arguments. There are workarounds implemented where Immutables try to consult imports or parse/leave types unchanged from the source representation. In your example workaround didn't work for non-abstract accessor (default with initializer body), so this is what possible to fix, and judging from tests it worked.
